pkg/bot/discord.go

```diff
@@ -0,0 +1,133 @@
+package bot
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/infracloudio/botkube/pkg/config"
+	"github.com/infracloudio/botkube/pkg/execute"
+	"github.com/infracloudio/botkube/pkg/logging"
+)
+
+// DiscordBot listens for user's message, execute commands and sends back the response
+type DiscordBot struct {
+	Token          string
+	AllowKubectl   bool
+	RestrictAccess bool
+	ClusterName    string
+	ChannelID      string
+	BotID          string
+}
+
+// discordMessage contains message details to execute command and send back the result
+type discordMessage struct {
+	Event         *discordgo.MessageCreate
+	BotID         string
+	Request       string
+	Response      string
+	IsAuthChannel bool
+	Session       *discordgo.Session
+}
+
+// NewDiscordBot returns new Bot object
+func NewDiscordBot() Bot {
+	c, err := config.New()
```

Comment:
better to pass config as a parameter to NewDiscordBot() instead of creating new instance each time. Can be handled in https://github.com/infracloudio/botkube/issues/270

Reply:
can it be considered for refactoring #270? as this same condition prevails across all the bots

Reply:
@codenio We have fixed this for other Bots with https://github.com/infracloudio/botkube/pull/272. Could you please address this as a part of this PR only?

Reply:
@PrasadG193 address those changes and updated the PR yesterday, can you recheck once